apps/cowswap-frontend/src/modules/swap/helpers/getAmountsForSignature.ts

```diff
@@ -0,0 +1,73 @@
+import { ONE_FRACTION } from '@cowprotocol/common-const'
+import { OrderKind } from '@cowprotocol/cow-sdk'
+import { Currency, CurrencyAmount, Percent, TradeType } from '@uniswap/sdk-core'
+
+import TradeGp from 'legacy/state/swap/TradeGp'
+
+interface AmountForSignatureParams {
+  trade: TradeGp
+  allowedSlippage: Percent
+  kind: OrderKind
+  featureFlags: {
+    swapZeroFee: boolean | undefined
+  }
+}
+
+function inputAmountForSignature(params: AmountForSignatureParams): CurrencyAmount<Currency> {
+  const { trade, allowedSlippage, kind, featureFlags } = params
+  const fee = trade.fee.feeAsCurrency
+  const slippageCoeff = ONE_FRACTION.add(allowedSlippage)
+
+  if (featureFlags.swapZeroFee) {
+    if (kind === OrderKind.SELL) {
+      return trade.inputAmount
+    } else {
+      return trade.inputAmountWithoutFee
+        .multiply(slippageCoeff) // add slippage
+        .add(fee) // add fee
+    }
+  }
+
+  if (kind === OrderKind.SELL) {
+    return trade.inputAmountWithFee
+  } else {
+    return trade.inputAmountWithFee.multiply(slippageCoeff) // subtract slippage
```

Comment:
The wording in the comment is confusing me.
This is the inputAmount for a buy order, which _adds_ slippage on top or the regular amount.


```suggestion
    return trade.inputAmountWithFee.multiply(slippageCoeff) // add slippage
```